lib/block-supports/layout.php

```diff
@@ -64,6 +65,17 @@ function gutenberg_get_layout_style( $selector, $layout, $has_block_gap_support
 			$style .= "$selector .alignfull { max-width: none; }";
 		}
 
+		// Handle negative margins for alignfull children of blocks with custom padding set.
+		// They're added separately because padding might only be set on one side.
+		if ( isset( $block_padding['right'] ) ) {
```

Comment:
It should work whenever a block has content width set, whether it's through `inherit` or setting an explicit content width on the block. Theoretically it's already being output only when needed, because blocks _without_ content width won't even go through this layout logic. We shouldn't need them to be full width either, but I'll re-test tomorrow to see if I missed anything!

Reply:
Oh wait you're right, this bit will always run even if the block has no content width. Makes sense to put in a condition then.